### PR TITLE
[clang][ASTImport] fix issue on anonymous enum import

### PR DESCRIPTION
**Pull request by jcsxky**
Don't skip searching in `ToContext` during importing `EnumDecl`. And `IsStructuralMatch` in `StructralEquivalence` can make sure to determine whether the found result is match or not.

### CODE REVIEW

**Review by balazske — COMMENTED** [review]
Does this work on the following code?
```
     struct A {
        enum { E1,E2 } x;
        enum { E3,E4 } y;
      };
 ```

**Comment by jcsxky** [comment]
> Does this work on the following code?
> 
> ```
>      struct A {
>         enum { E1,E2 } x;
>         enum { E3,E4 } y;
>       };
> ```

I have updated testcase according to your suggestion to demonstrate this solution works correctly on the testcase above.

File: clang/unittests/AST/ASTImporterTest.cpp [hunk 1]
```diff
@@ -9674,6 +9674,27 @@ TEST_P(ASTImporterOptionSpecificTestBase, ImportInstantiatedFromMember) {
...
+      R"(
+      struct A {
+        enum { E1,E2} x;
+      };
+      )";
+  Decl *FromTU = getTuDecl(Code, Lang_CXX11);
+  auto *FromE = FirstDeclMatcher<EnumDecl>().match(FromTU, enumDecl());
+  auto *ToE = Import(FromE, Lang_CXX11);
+  ASSERT_TRUE(ToE);
+  EXPECT_FALSE(ToE->enumerators().empty());
```
[the remark this answers is not in the document]
Comment (jcsxky) [reply clang/unittests/AST/ASTImporterTest.cpp]: Done.

**Review by balazske — COMMENTED** [review]
The change looks good, only the test can be made more exact.

File: clang/unittests/AST/ASTImporterTest.cpp [hunk 2]
```diff
@@ -9674,6 +9674,40 @@ TEST_P(ASTImporterOptionSpecificTestBase, ImportInstantiatedFromMember) {
...
+  const char *ToCode =
+      R"(
+      struct A {
+        enum { E1, E2} x;
+        enum { E3, E4} y;
+      };
+      )";
+  Decl *ToTu = getToTuDecl(ToCode, Lang_CXX11);
+  auto *ToE = FirstDeclMatcher<EnumDecl>().match(
+      ToTu, enumDecl(hasEnumConstName("E1")));
```
[the remark this answers is not in the document]
Comment (jcsxky) [reply clang/unittests/AST/ASTImporterTest.cpp]: Import test on the other enum has been added to make this test completely.

**Review by balazske — APPROVED** [review]

**Pull request merged**